compiler/rustc_mir_transform/src/coverage/spans/from_mir.rs

```diff
@@ -180,14 +180,14 @@ fn filtered_terminator_span(terminator: &Terminator<'_>) -> Option<Span> {
 /// Returns an extrapolated span (pre-expansion[^1]) corresponding to a range
 /// within the function's body source. This span is guaranteed to be contained
 /// within, or equal to, the `body_span`. If the extrapolated span is not
-/// contained within the `body_span`, the `body_span` is returned.
+/// contained within the `body_span`, `None` is returned.
 ///
 /// [^1]Expansions result from Rust syntax including macros, syntactic sugar,
 /// etc.).
 #[inline]
-fn function_source_span(span: Span, body_span: Span) -> Span {
+fn unexpand_into_body_span(span: Span, body_span: Span) -> Option<Span> {
     use rustc_span::source_map::original_sp;
 
     let original_span = original_sp(span, body_span).with_ctxt(body_span.ctxt());
-    if body_span.contains(original_span) { original_span } else { body_span }
+    body_span.contains(original_span).then_some(original_span)
 }
```

Comment:
This function looks a lot like `Span::find_ancestor_inside`. Should it be reused? Or is there a subtle difference?

Reply:
I was mostly just modifying the existing code, and didn't know about `Span::find_ancestor_inside` as a potential option.

I tried switching over, and the results are *mostly* the same, with a few small differences in some corner cases.

At a glance the changes seem like they might be improvements, but I'm not confident in that conclusion. What I might do is stick with the status quo (`original_sp`) for now, but leave a FIXME to remind me to investigate switching over as a separate PR.